tools/cli/domainCommands.go

```diff
@@ -353,6 +353,38 @@ func (d *domainCLIImpl) failoverDomains(c *cli.Context) ([]string, []string) {
 	return succeedDomains, failedDomains
 }
 
+// RebalanceDomains is used for managed failover to rebalance with domain data PreferredCluster
+func (d *domainCLIImpl) RebalanceDomains(c *cli.Context) {
+	// ask user for confirmation
+	prompt("You are trying to rebalance all managed domains, continue? Y/N")
+	d.rebalanceDomains(c)
+}
+
+func (d *domainCLIImpl) rebalanceDomains(c *cli.Context) ([]string, []string) {
+	domains := d.getAllDomains(c)
+	shouldRebalance := func(domain *types.DescribeDomainResponse) bool {
+		return len(getPreferredClusterName(domain)) != 0 && isDomainFailoverManagedByCadence(domain)
+	}
+	var succeedDomains []string
+	var failedDomains []string
+	for _, domain := range domains {
+		if shouldRebalance(domain) {
+			domainName := domain.GetDomainInfo().GetName()
+			err := d.failover(c, domainName, getPreferredClusterName(domain))
```

Comment:
Just want to check my understanding: domain cache is refreshed every 10s so all failover requests happens within the 10 seconds will take affect together and only one failover queue per shard will be created?

If too many domain need to rebalanced, do we need to do them in batches?

Reply:
Most likely it can finish in 10s so at most two failover queue will be created